cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MetadataDataset.java

```diff
@@ -397,46 +398,48 @@ private Metadata getSnapshotBeforeTime(Id.NamespacedId targetId, long timeMillis
    */
   public List<MetadataEntry> search(final String namespaceId, final String searchQuery,
                                     final MetadataSearchTargetType type) {
-    Set<MetadataEntry> results = new HashSet<>();
+    List<MetadataEntry> results = new ArrayList<>();
```

Comment:
We index them only once. But here in the search we are doing search for every word and it's very much possible that we can see same word again in search query specially since now we support multi word search query. This PR also introduces a basic scoring where entities which match most words in search queries are ranked higher. I agree we shouldn't over optimize here as its all going to go away but if we just do search for unique words in query as you mentioned too we will not be searching for same words in query again and again and also the scoring will not get affected by it.